tests/system/providers/amazon/aws/example_sagemaker.py

```diff
@@ -161,8 +161,12 @@ def _build_and_upload_docker_image(preprocess_script, repository_uri):
 
         docker_build_and_push_commands = f"""
             cp /root/.aws/credentials /tmp/credentials &&
+            # login to public ecr repo containing amazonlinux image
+            docker login --username {username} --password {password} public.ecr.aws
             docker build --platform=linux/amd64 -f {dockerfile.name} -t {repository_uri} /tmp &&
             rm /tmp/credentials &&
+
+            # login again, this time to the private repo we created to hold that specific image
             aws ecr get-login-password --region {ecr_region} |
             docker login --username {username} --password {password} {repository_uri} &&
             docker push {repository_uri}
```

Comment:
BTW, might be useful as follow up. About month ago I added for further integration with docker EcrHook (Private)

https://github.com/apache/airflow/blob/bfbe2cb6fc39ccc24a13befa28e63beccbbf37b0/airflow/providers/amazon/aws/hooks/ecr.py#L67-L78

Potentially you could obtain credentials as

```python
creds = EcrHook(aws_conn_id=None, region_name=ecr_region).get_temporary_credentials()[0]

```

Reply:
And potentially we could create almost the same hook for public ECR, since we know why it could be useful

Reply:
oh yes nice this pushes the weird logic of decoding the credentials out of here which is good.
However I don't think we need anything special for the public ECR since it's using the same credentials.

Reply:
I mean if I finally 🤣 🤣  complete integrate DockerHook with custom Auth method, such as ECR (regular) it might be useful if some one need only ECR Public.